File: ts-sdk/src/services/audit.ts
import { BizRecord } from '../layers/domain';
import { SecRecord } from '../layers/sec';

export class AuditService {
  async writeAuditRecord(record: BizRecord | SecRecord): Promise<void> {
    const entry = record.toDict();
    entry._audit_signature = "sha256:ts-signed";
    
    // In browser, this would typically be a secure POST
    console.warn('[AUDIT] Immutable WORM Write (Simulated)', entry);
  }
}

File: py-sdk/mylogger/services/audit.py
from typing import Union
from ..layers.biz import BizRecord
from ..layers.sec import SecRecord

class AuditService:
    def __init__(self, storage_path: str = "./logs/audit.log"):
        self.path = storage_path

    def write_audit_record(self, record: Union[BizRecord, SecRecord]):
        # Simulated WORM (Write Once Read Many) storage
        # Real implementation would use HMAC signing
        entry = record.to_dict()
        entry["_audit_signature"] = "sha256:..." 
        with open(self.path, "a") as f:
            import json
            f.write(json.dumps(entry) + "\n")
